adyenv6b2ccheckoutaddon/acceleratoraddon/web/src/com/adyen/v6/controllers/pages/AdyenComponentController.java

```diff
@@ -0,0 +1,205 @@
+import javax.servlet.http.HttpServletResponse;
+import javax.validation.Valid;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+import static com.adyen.v6.constants.AdyenControllerConstants.COMPONENT_PREFIX;
+
+@Controller
+@RequestMapping(COMPONENT_PREFIX)
+public class AdyenComponentController {
+    private static final Logger LOGGER = Logger.getLogger(AdyenComponentController.class);
+
+    @Resource(name = "adyenCheckoutFacade")
+    private AdyenCheckoutFacade adyenCheckoutFacade;
+
+    @Resource(name = "checkoutFlowFacade")
+    private CheckoutFlowFacade checkoutFlowFacade;
+
+    @Resource(name = "acceleratorCheckoutFacade")
+    private AcceleratorCheckoutFacade checkoutFacade;
+
+    @RequestMapping(value = "/payment", method = RequestMethod.POST)
+    @ResponseBody
+    public String componentPayment(final Model model,
+                                final HttpServletRequest request,
+                                @Valid final AdyenPaymentForm adyenPaymentForm,
+                                final BindingResult bindingResult) throws AdyenComponentException {
+        LOGGER.debug("Component paymentForm: " + adyenPaymentForm);
+
+        //Save payment information
+        getAdyenCheckoutFacade().handlePaymentForm(adyenPaymentForm, bindingResult);
+        if (bindingResult.hasGlobalErrors() || bindingResult.hasErrors()) {
+            LOGGER.debug(bindingResult.getAllErrors().stream().map(DefaultMessageSourceResolvable::getCode).reduce((x, y) -> (x = x + y)));
+            if (adyenPaymentForm.getBillingAddress() != null) {
+                adyenPaymentForm.resetFormExceptBillingAddress();
+            }
+            throw new AdyenComponentException("checkout.error.paymentethod.formentry.invalid");
+        }
+
+        try {
+            validateOrderForm();
+
+            //Make payment request
+            Gson gson = new GsonBuilder().disableHtmlEscaping().create();
+            String payPalDetailsJson = adyenPaymentForm.getComponentData();
+            PayPalDetails payPalDetails = gson.fromJson(payPalDetailsJson, PayPalDetails.class);
+
+            final CartData cartData = getCheckoutFlowFacade().getCheckoutCart();
+
+            PaymentsResponse paymentsResponse = getAdyenCheckoutFacade().componentPayment(request, cartData, payPalDetails);
+            return gson.toJson(paymentsResponse);
+        } catch (InvalidCartException e) {
+            LOGGER.error("InvalidCartException", e);
+            throw new AdyenComponentException(e.getMessage());
+        } catch (AdyenNonAuthorizedPaymentException | ApiException e) {
+            LOGGER.error("PaymentException", e);
+            throw new AdyenComponentException("checkout.error.authorization.payment.refused");
+        } catch (Exception e) {
+            LOGGER.error("Exception", e);
+            throw new AdyenComponentException("checkout.error.authorization.payment.error");
```

Comment:
https://github.com/Adyen/adyen-hybris/blob/262e2312c03643353401dd3c19c15b0c860bc6ad/adyenv6b2ccheckoutaddon/acceleratoraddon/web/webroot/WEB-INF/messages/base.properties#L8
Can you please fix the typo related to "checkout.error.authorization.payment.error" above? "occured" -> "occurRed"

Reply:
Fixed

Reply:
:)